app/models/post.rb

```diff
@@ -1,34 +1,14 @@
 class Post < ActiveRecord::Base
-  belongs_to :user
   has_and_belongs_to_many :tags
   has_many :comments, as: :parent
 
   validates_presence_of :title
-  validates_presence_of :user, :unless => :child?
 
-  before_save do
-    unless child?
-      parser = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true)
-      self.body_html = parser.render(body_markdown)
-    end
-  end
-
-  # after_create :notify_twitter
-  # TODO: move this to background job
-  def notify_twitter
-    $twitter_client.update(tweet_content)
-  end
+  scope :internalpost, -> { where(type: 'InternalPost') }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.